hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeyCommitRequest.java

```diff
@@ -248,7 +259,7 @@ public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager,
       long correctedSpace = omKeyInfo.getReplicatedSize();
       // if keyToDelete isn't null, usedNamespace needn't check and
       // increase.
-      if (keyToDelete != null && isHSync) {
+      if (keyToDelete != null && (isHSync || isPreviousCommitHsync)) {
         correctedSpace -= keyToDelete.getReplicatedSize();
         checkBucketQuotaInBytes(omMetadataManager, omBucketInfo,
             correctedSpace);
```

Comment:
If there are two clients opening the same key (two entries in `OpenKeyTable` with the same key path, but different `clientId` suffix) and perform hsync, will the blocks still get correctly discarded with this change?

Currently the logic doesn't prevent two clients from hsync'ing on the same key.

Reply:
In normal scenario, when another client performs commit, it do overwrite the previous client commit and delete its block. But For Hsync, this scenario can cause dataloss and this needs to be handled. Created another JIRA to track this scenario,
https://issues.apache.org/jira/browse/HDDS-9176